app/serializers/hash_with_indifferent_access_serializer.rb

```diff
@@ -0,0 +1,10 @@
+# Based on: https://nandovieira.com/using-postgresql-and-jsonb-with-ruby-on-rails
+class HashWithIndifferentAccessSerializer
```

Comment:
Not really sure what the purpose of this class is? It feels like this should have a more rigidly defined structure so that we can have appropriate swagger documentation, schemas, specs etc.

Reply:
so, this serializer makes jsonb objects behave as hashes with indifferent access, to make it easier to work with.
The `details` field in an event contains the full event parameters used to create the event, which will vary from event to event.  A cancellation event must specify a cancellation_reason for example, whereas a redirect event must specify a `to_location` relationship. So a rigid structure with a field for to_location or a field for cancellation_reason is not a good fit in this case (IMHO).